py3status/i3status.py

```diff
@@ -195,19 +195,29 @@ def __init__(self, py3_wrapper):
         self.standalone = py3_wrapper.config['standalone']
         self.time_modules = []
         self.tmpfile_path = None
+        self.update_due = 0
 
     def update_times(self):
         """
         Update time for any i3status time/tztime items.
+        Returns the time till next update needed.
         """
-        updated = []
-        for module in self.i3modules.values():
-            if module.is_time_module:
-                if module.update_time_value():
-                    updated.append(module.module_name)
-        if updated:
-            # trigger the update so new time is shown
-            self.py3_wrapper.notify_update(updated)
+        now = time()
+        if now > self.update_due:
+            updated = []
+            for module in self.i3modules.values():
+                if module.is_time_module:
+                    if module.update_time_value():
+                        updated.append(module.module_name)
+            if updated:
+                # trigger the update so new time is shown
+                self.py3_wrapper.notify_update(updated)
+                # time we next need to do an update
+            self.update_due = int(now) + 1
```

Comment:
This is because we need to update the i3status `time` module seconds.

This needs to be fixed properly but doing this just keeps things working without confusing this PR .

Once this is merged I can start fixing that up.  But it needs a fair bit of refactoring of `i3status.py`.